processing/src/main/java/org/apache/carbondata/processing/surrogatekeysgenerator/csvbased/BadRecordsLogger.java

```diff
@@ -179,6 +179,7 @@ private synchronized void writeBadRecordsToFile(StringBuilder logStrings) {
 
       }
       bufferedWriter.write(logStrings.toString());
+      bufferedWriter.flush();
```

Comment:
It is not the correct way, please don't flush for every row. it slows down a lot.
 Please close the streams properly, it will solve the problem